pwndbg/chain.py

```diff
@@ -143,7 +144,7 @@ def format(
     arrow_right = c.arrow(f" {config_arrow_right} ")
 
     # Colorize the chain
-    rest = [M.get_address_and_symbol(link) for link in chain]
+    rest = [M.get_address_and_symbol(link, display_heap_free_bins) for link in chain]
```

Comment:
Should we maybe handle such custom detecting/resolving of "nice things to enhance" during telescope/chain resolution in a more generic manner?

I mean, lets assume we want to enhance "errno values" one day, would we also add an additional argument here? :/

Perhaps there should be a way to register/enable/disable such different/additional handlers for formatting/telescope stuff? Not sure

Reply:
(I'm okay with merging this as is, as long as we resolve other comments, but perhaps this is a good time to do it properly to handle future stuff now?)

Reply:
Yes - that's a good point. While making this function parameter, I also had the thought "well, what if one day we want to display floating point numbers in this enhance function too"... All the different arguments will add up, especially since we need to pass them through two functions.

Ideas:
- Some sort of detection for each number - check if it's in the heap area, or check if it resolves to a floating point number that's reasonable (which misses cases, and for things like errno, this gets harder as we need to remember we are from the `RAX` register)
- Pass all the additional info in a single struct instead of separate parameters
- set_telescope_handler(ENUM, True/False) - enum to set a value globally that telescope checks?
   -  or maybe a function callback instead of enum - it takes a list of functions that are chained like a plugin system, all are called to possibly enhance the value (or return None if deemed irrelevent)

Reply:
> this gets harder as we need to remember we are from the RAX register

We could maybe pass in some context around (?) tho I am not sure how practical that is in general. I am not 100% bought or enhancing the errno yet, but if we find a nice solution we could do that.

> set_telescope_handler(ENUM, True/False) - enum to set a value globally that telescope checks?

Could also be sth like `set telescope-enhance x,y,z` or sth.

Reply:
`set telescope-enhance free-bins` and `set telescope-enhance free-bins floating errno` to enable globally sounds good to me. We still need the caller to the telescope function to indicate that "in this specific case, don't do free-bins (regardless of global setting)".

Perhaps a context object would be good - a optionally `None` parameter that provides additional data relevant to the current telescope. Could contain "register: RAX", or "heap: True", "floating: True". This would avoid having to add a new parameter each time we want some new logic. I'll consider this and a `enhancer callback function` option and see which one makes most sense.